enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  ## my_all method
+  def my_all
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_none method
+  def my_none
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_count method
+  def my_count
+    return self unless block_given?
+
+    arr = self
+    counter = 0
+    arr.my_each { |i| counter += 1 if yield(i) }
+    counter
+  end
```

Comment:
- [ ]  enumerables #my_count returns the number of items in enum through enumeratio
- [ ] enumerables #my_count returns the number of items in enum through enumeration::range
    Error:  expect(range.my_count).to eq range.count
- [ ] enumerables #my_count counts the number of items in enum that are equal to item if an argument is given
      Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)